### DISCUSSION TITLE
경🎉 디스커션 오픈 🎉축

### DISCUSSION THREAD

**Discussion by devitssu** [one-day-one-meal/one-hour-one-meal#39] [Show and tell]
**질문**, **도움요청**, 자유로운 **의견 공유** 등은 이슈보다는 여기서하면 될 거 같아요.
저도 처음 써봐요 🕺

**Comment by mobzzzzz** [comment]
대토론의 장이 열렸군요! 고생하셨습니다 :+1:

**Comment by Moo-moo-11** [comment]
우왕 :laughing:

**Comment by Moo-moo-11** [comment]
data class CourseResponse(
    val id: Long,
    val chefId: Long,
    val title: String,
    val describe: String,
    val status: String,
    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?
)
현재 CourseResponse를 이렇게 작성해두었는데 chefId를 반환하는 것이 괜찮은 걸까요? User 테이블에서 nickname을 조회해서 응답할까요?

**Reply by mobzzzzz** [reply]
프론트에서 데이터 받을 때 이름을 위해 추가로 chefId를 조회해야 하면 프론트에서 번거로움이 커지긴 합니다!
무준님 말씀대로 서비스 로직에서 user를 조회해서 nickname 이나 chefName 처럼 명시적인 chef의 이름을 Dto에 보관해서 같이 보내주는게 좋다고 생각합니다.

**Reply by devitssu** [reply]
아니면 UserResponse 객체를 담아주셔도 좋을것 같아요.
만약 프론트에서 강의를 만든 셰프를 클릭하면 프로필을 볼 수 있게 하는 기능을 추가한다고 하면 UserResponse에 담긴 userId로 조회할 수 있으니까요

**Comment by Moo-moo-11** [comment]
좋아요 기능이 API 상에서 /api/v1/courses/{courseId}/thumbs-up 으로 POST, DELETE로 매핑되어 있어서 일단 CourseController와 CourseService에 해당 함수들은 만들어 놓았는데 괜찮을까요? 아예 Course 모델 패키지 안에 Like 테이블을 만들까요 아니면 Like 패키지를 따로 분리시키고 LikeService를 따로 만들까요?

**Reply by mobzzzzz** [reply]
좋아요 기능 자체가 완전히 Course에만 의존적인 서비스라서 CourseController, CourseService에 포함하는 건 좋은 것 같고
이에 따라 ThumbsUp 모델이  Course 패키지에 있는 것도 충분히 설명 가능할 것 같습니다!
